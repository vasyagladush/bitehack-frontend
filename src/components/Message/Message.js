import React from 'react';

const message = props =>{
    return(
        <div style={{marginTop: 15}}>
        {props.children}
        </div>
    )
}

export default message;