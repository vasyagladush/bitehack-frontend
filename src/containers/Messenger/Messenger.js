// import React,{useState, useEffect,useRef} from 'react';
// import Message from '../../components/Message/Message';
// import classes from './Messenger.module.css';
// import Emoji from '../../components/Emoji/Emoji';
// import {Form,Image} from 'react-bootstrap'; 
// import axios from 'axios';
// import {connect} from 'react-redux';
// import openSocket from 'socket.io-client';
// import {url} from '../../ApiUrl';
// import * as actions from '../../store/actions/index';

// // const socket = openSocket(url,{transports: ['websocket']});
// const Messenger = props =>{
//     const [input,setInput] = useState('');
//     const messagesEndRef = useRef(null);
//     const scrollToBottom = ()=>{
//         messagesEndRef.current.scrollIntoView({ behavior: "smooth" });
//     }
//     const submitMessage = async event =>{
//         event.preventDefault();
//         if (input === '')
//         {
//             return null
//         }
//         const {data} = await axios.post(url+"/chat/message",{
//             personId: props.personId, 
//             message: input
//         },{headers:{
//             'Authorization': 'Bearer '+localStorage.getItem("token")
//         }});

//         console.log(data);
//         // axios.post(url+'/user/message',{
//         //  id: props.personId,
//         //  message: input 
//         // },{
//         //     headers: {
//         //         'Authorization': 'Bearer '+localStorage.getItem('token')
//         //     }
//         // })
//         // .then(response =>{
//         //     props.addMessage(response.data);
//         //     setInput('');
//         // })
//         // .catch(err=>{
//         //     console.log(err);
//         // })
//     }
//     const handlingInput = event =>{
//         setInput(event.target.value);
//     }

//     useEffect(()=>{
//         console.log(props.messages);
//         // socket.emit('connect');
//         // socket.on('message',(data)=>{
//         //         if(props.personId === data.person._id && data.personGetMessage.toString()===props.meId.toString())
//         //         {
//         //         props.addMessage(data);
//         //         setInput('');
//         //         }
//         // });
//         // return ()=>socket.off('message');
//     },[]);

//     useEffect(scrollToBottom,[props.messages]); // scroll if our messages are changed

//     // let messages = null;
//     // if (props.messages.length){
//     //     messages= props.messages.map(message=>{
//     //         return (
//     //             <Message key={message._id}>
//     //                 <p style={{fontWeight: 'bold', display: 'inline'}}>{message.person.name}:</p> {message.message}
//     //             </Message>
//     //         )
//     //     })
//     // }
//     // else
//     // {
//     //     messages = (<p>Type something !<Emoji symbol="âœ‹"/></p>)
//     // }
//     return(
//         <div style={{
//             borderWidth:'1px',
//             borderColor: 'black'
//         }}>
//             <div className={classes.messenger}>
//                 {props.messages}
//                 <div ref={messagesEndRef}/> {
//                     //smooth
//                 }
//             </div>
//                 {/* <Form className={props.personId ? classes.form : classes.formunactive} onSubmit={submitMessage}>
//                     <Form.Group controlId="formBasicEmail">
//                         <Form.Control type="text" placeholder="Enter Some text" className={classes.input} name="textInput" value={input} onChange={handlingInput}/>
//                     </Form.Group>
//                 </Form> */}
//                 <input type="text" placeholder='Enter something' value={input} onChange={handlingInput}/>

//                 <Image src={require('../../images/send.png')} className={classes.image} onClick={submitMessage}/>
//         </div>

//     )
// };
// // const mapStateToProps = state =>{
// //     return {
// //         personId: state.personId,
// //         messages: state.messages,
// //         meId: state.meId
// //     }
// // };

// // const mapDispatchToProps = dispatch=>{
// //     return {
// //         setMessages: (messages)=>dispatch(actions.setMessages(messages)),
// //         addMessage: (data)=>dispatch(actions.addMessage(data)),
// //         setPerson: (id)=>dispatch(actions.setUser(id))
// //     }
// // }

// export default Messenger; 