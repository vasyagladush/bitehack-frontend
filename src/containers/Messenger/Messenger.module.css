.messenger { 
    top: 10%;
    left: 25%;
    width: 50%;
    height: 60%;
    border: 1px solid black;
    overflow: auto;
    position: absolute;
}

.formunactive {
    pointer-events: none;
    position: absolute;
    top: 70%;
    left: 25%;
    width: 50%;
    margin-top: 10px;
}

.form {
    position: absolute;
    top: 70%;
    left: 25%;
    width: 50%;
    margin-top: 10px;
}

.image { 
    width: 30px;
    height: 30px;
    position: absolute;
    top: 72%;
    left: 76%;
}

.image:hover {
    cursor: pointer;
}