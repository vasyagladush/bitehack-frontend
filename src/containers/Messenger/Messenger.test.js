import React from 'react';

import {Provider} from 'react-redux';
import {Form,Image} from 'react-bootstrap';
import Messenger from './Messenger';
import Message from '../../components/Message/Message';
import Adapter from 'enzyme-adapter-react-16';
import {configure,mount} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

let store = configureMockStore()({    messages: [{
    _id: 'dummy',
    person: {
        name: 'siema',
        surname: 'elo'
    },
    message: 'elooooooo'
}]});
configure({adapter: new Adapter});

describe('<Messenger />', ()=>{
    let wrapper;
    beforeEach(()=>{
        window.HTMLElement.prototype.scrollIntoView = function() {};
        wrapper = mount(
        <Provider store={store}>
            <Messenger/>
        </Provider>
        );
    });
    it('matches snapshot',()=>{
        wrapper = renderer.create(
            <Provider store={store}>
            <Messenger/>
        </Provider>
        ).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('should render message',()=>{
        expect(wrapper.find(Message)).toHaveLength(1);
    });
    it('should render Form.Control',()=>{
        expect(wrapper.find(Form.Control)).toHaveLength(1);
    });
    it('should render image which means sending message',()=>{
        expect(wrapper.find(Image)).toHaveLength(1);
    }); 
    it("should render paragraph if we don't have messages",()=>{
        store = configureMockStore()({messages: []});
        wrapper = mount(
            <Provider store={store}>
            <Messenger/>
        </Provider>
        );
        expect(wrapper.find('p').text()).toEqual('Type something !âœ‹');
    });
});