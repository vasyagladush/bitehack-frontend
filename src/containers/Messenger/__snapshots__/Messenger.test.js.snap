// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Messenger /> matches snapshot 1`] = `
<div
  style={
    Object {
      "borderColor": "black",
      "borderWidth": "1px",
    }
  }
>
  <div
    className="messenger"
  >
    <div
      style={
        Object {
          "marginTop": 15,
        }
      }
    >
      <p
        style={
          Object {
            "display": "inline",
            "fontWeight": "bold",
          }
        }
      >
        siema
        :
      </p>
       
      elooooooo
    </div>
    <div />
     
  </div>
  <form
    className="formunactive"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        className="input form-control"
        id="formBasicEmail"
        name="textInput"
        onChange={[Function]}
        placeholder="Enter Some text"
        type="text"
        value=""
      />
    </div>
  </form>
  <img
    className="image"
    onClick={[Function]}
    src="send.png"
  />
</div>
`;
